From 84e3f64cfd0ee83d43626d9244cae225b9f925b3 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Thu, 9 Jan 2020 13:36:50 -0800
Subject: [PATCH] Fix build with gtest-1.8.1

* taken from:
  https://github.com/googlecartographer/cartographer/issues/1611#issuecomment-560236725

* linking fails without this:
/usr/include/gmock/gmock-matchers.h:2059: error: undefined reference to 'testing::AssertionSuccess()'
/usr/include/gmock/gmock-matchers.h:2066: error: undefined reference to 'testing::AssertionFailure()'
/usr/include/gmock/gmock-matchers.h:2066: error: undefined reference to 'testing::AssertionResult::AssertionResult(testing::AssertionResult const&)'
/usr/include/gtest/gtest-message.h:247: error: undefined reference to 'testing::Message::GetString[abi:cxx11]() const'
/usr/include/gtest/gtest-printers.h:155: error: undefined reference to 'testing::internal2::PrintBytesInObjectTo(unsigned char const*, unsigned int, std::ostream*)'
/usr/include/gtest/gtest-printers.h:616: error: undefined reference to 'testing::internal::PrintStringTo(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::ostream*)'
/usr/include/gtest/gtest.h:1435: error: undefined reference to 'testing::internal::EqFailure(char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
/usr/include/gtest/gtest.h:1445: error: undefined reference to 'testing::AssertionSuccess()'
/usr/include/gtest/gtest.h:335: error: undefined reference to 'testing::Message::Message()'
/usr/include/gtest/gtest.h:352: error: undefined reference to 'testing::Message::GetString[abi:cxx11]() const'
/usr/include/gtest/internal/gtest-internal.h:664: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/include/gtest/internal/gtest-port.h:1215: error: undefined reference to 'testing::internal::IsTrue(bool)'
/usr/include/gtest/internal/gtest-port.h:2061: error: undefined reference to 'testing::internal::GTestLog::GTestLog(testing::internal::GTestLogSeverity, char const*, int)'
/usr/include/gtest/internal/gtest-port.h:2061: error: undefined reference to 'testing::internal::GTestLog::~GTestLog()'
/usr/include/gtest/internal/gtest-port.h:2061: error: undefined reference to 'testing::internal::g_linked_ptr_mutex'
/usr/include/gtest/internal/gtest-port.h:2073: error: undefined reference to 'testing::internal::GTestLog::GTestLog(testing::internal::GTestLogSeverity, char const*, int)'
/usr/include/gtest/internal/gtest-port.h:2073: error: undefined reference to 'testing::internal::GTestLog::~GTestLog()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:103: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:103: error: undefined reference to 'testing::Test::~Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:103: error: undefined reference to 'testing::internal::GetTestTypeId()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:103: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:37: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:37: error: undefined reference to 'testing::Test::~Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:37: error: undefined reference to 'testing::internal::GetTestTypeId()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:37: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:88: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:88: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:88: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:88: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:91: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:91: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:91: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:91: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:92: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:92: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:92: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:92: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:93: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:93: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:93: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/collator_test.cc:93: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:29: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:29: error: undefined reference to 'testing::Test::~Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:29: error: undefined reference to 'testing::internal::GetTestTypeId()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:29: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:34: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:34: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:34: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:34: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:38: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:38: error: undefined reference to 'testing::Test::~Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:38: error: undefined reference to 'testing::internal::GetTestTypeId()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:38: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:43: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:43: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:43: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:43: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:47: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:47: error: undefined reference to 'testing::internal::GetTestTypeId()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:47: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:52: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:52: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:52: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/internal/voxel_filter_test.cc:52: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:40: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:40: error: undefined reference to 'testing::Test::~Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:40: error: undefined reference to 'testing::internal::GetTestTypeId()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:40: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:54: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:54: error: undefined reference to 'testing::Test::~Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:54: error: undefined reference to 'testing::internal::GetTestTypeId()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:54: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:65: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:65: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:65: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:65: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:65: error: undefined reference to 'testing::internal::GetBoolAssertionFailureMessage[abi:cxx11](testing::AssertionResult const&, char const*, char const*, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:73: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:73: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:73: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:73: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:73: error: undefined reference to 'testing::internal::GetBoolAssertionFailureMessage[abi:cxx11](testing::AssertionResult const&, char const*, char const*, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:79: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:79: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:79: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:79: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:82: error: undefined reference to 'testing::internal::GetBoolAssertionFailureMessage[abi:cxx11](testing::AssertionResult const&, char const*, char const*, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:91: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:91: error: undefined reference to 'testing::internal::GetTestTypeId()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:91: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:97: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:97: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:97: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:97: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/map_by_time_test.cc:97: error: undefined reference to 'testing::internal::GetBoolAssertionFailureMessage[abi:cxx11](testing::AssertionResult const&, char const*, char const*, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:28: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:28: error: undefined reference to 'testing::Test::~Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:28: error: undefined reference to 'testing::internal::GetTestTypeId()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:28: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:40: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:40: error: undefined reference to 'testing::Test::~Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:40: error: undefined reference to 'testing::internal::GetTestTypeId()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:40: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:46: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:46: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:46: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:46: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:46: error: undefined reference to 'testing::internal::DoubleNearPredFormat(char const*, char const*, char const*, double, double, double)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:47: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:47: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:47: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:47: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:47: error: undefined reference to 'testing::internal::DoubleNearPredFormat(char const*, char const*, char const*, double, double, double)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:48: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:48: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:48: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:48: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:48: error: undefined reference to 'testing::internal::DoubleNearPredFormat(char const*, char const*, char const*, double, double, double)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:49: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:49: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:49: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:49: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/sensor/point_cloud_test.cc:49: error: undefined reference to 'testing::internal::DoubleNearPredFormat(char const*, char const*, char const*, double, double, double)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:29: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:29: error: undefined reference to 'testing::Test::~Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:80: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:80: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:80: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:80: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:81: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:81: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:81: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:81: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:82: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:82: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:82: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:82: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:84: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:84: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:84: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test.cc:84: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/rigid_transform_test_helpers.h:42: error: undefined reference to 'testing::internal::FormatMatcherDescription(bool, char const*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:29: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:29: error: undefined reference to 'testing::Test::~Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:29: error: undefined reference to 'testing::internal::GetTestTypeId()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:29: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:38: error: undefined reference to 'testing::internal::GetBoolAssertionFailureMessage[abi:cxx11](testing::AssertionResult const&, char const*, char const*, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:39: error: undefined reference to 'testing::internal::GetBoolAssertionFailureMessage[abi:cxx11](testing::AssertionResult const&, char const*, char const*, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:40: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:40: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:40: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:40: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:40: error: undefined reference to 'testing::internal::GetBoolAssertionFailureMessage[abi:cxx11](testing::AssertionResult const&, char const*, char const*, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:41: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:41: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:41: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:41: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:41: error: undefined reference to 'testing::internal::GetBoolAssertionFailureMessage[abi:cxx11](testing::AssertionResult const&, char const*, char const*, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:42: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:42: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:42: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:42: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:43: error: undefined reference to 'testing::Message::Message()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:43: error: undefined reference to 'testing::internal::AssertHelper::AssertHelper(testing::TestPartResult::Type, char const*, int, char const*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:43: error: undefined reference to 'testing::internal::AssertHelper::operator=(testing::Message const&) const'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:43: error: undefined reference to 'testing::internal::AssertHelper::~AssertHelper()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:46: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:46: error: undefined reference to 'testing::Test::~Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:46: error: undefined reference to 'testing::internal::GetTestTypeId()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:46: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:65: error: undefined reference to 'testing::Test::Test()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:65: error: undefined reference to 'testing::internal::GetTestTypeId()'
/usr/src/debug/cartographer/1.0.0-1-r0/git/cartographer/transform/transform_interpolation_buffer_test.cc:65: error: undefined reference to 'testing::internal::MakeAndRegisterTestInfo(char const*, char const*, char const*, char const*, testing::internal::CodeLocation, void const*, void (*)(), void (*)(), testing::internal::TestFactoryBase*)'
/usr/src/debug/gtest/1.8.1-r0/git/googlemock/src/gmock_main.cc:52: error: undefined reference to 'testing::InitGoogleMock(int*, char**)'
/usr/src/debug/gtest/1.8.1-r0/git/googletest/include/gtest/gtest.h:2341: error: undefined reference to 'testing::UnitTest::GetInstance()'
/usr/src/debug/gtest/1.8.1-r0/git/googletest/include/gtest/gtest.h:2341: error: undefined reference to 'testing::UnitTest::Run()'
CMakeFiles/cartographer.sensor.internal.collator_test.dir/cartographer/sensor/internal/collator_test.cc.o:collator_test.cc:typeinfo for cartographer::sensor::(anonymous namespace)::Collator_OrderingMultipleTrajectories_Test: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.sensor.internal.collator_test.dir/cartographer/sensor/internal/collator_test.cc.o:collator_test.cc:typeinfo for cartographer::sensor::(anonymous namespace)::Collator_Ordering_Test: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.sensor.internal.collator_test.dir/cartographer/sensor/internal/collator_test.cc.o:collator_test.cc:vtable for cartographer::sensor::(anonymous namespace)::Collator_OrderingMultipleTrajectories_Test: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.sensor.internal.collator_test.dir/cartographer/sensor/internal/collator_test.cc.o:collator_test.cc:vtable for cartographer::sensor::(anonymous namespace)::Collator_OrderingMultipleTrajectories_Test: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.sensor.internal.collator_test.dir/cartographer/sensor/internal/collator_test.cc.o:collator_test.cc:vtable for cartographer::sensor::(anonymous namespace)::Collator_Ordering_Test: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.sensor.internal.collator_test.dir/cartographer/sensor/internal/collator_test.cc.o:collator_test.cc:vtable for cartographer::sensor::(anonymous namespace)::Collator_Ordering_Test: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.sensor.internal.voxel_filter_test.dir/cartographer/sensor/internal/voxel_filter_test.cc.o:voxel_filter_test.cc:typeinfo for cartographer::sensor::(anonymous namespace)::VoxelFilterTest_HandlesLargeCoordinates_Test: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.sensor.internal.voxel_filter_test.dir/cartographer/sensor/internal/voxel_filter_test.cc.o:voxel_filter_test.cc:typeinfo for cartographer::sensor::(anonymous namespace)::VoxelFilterTest_IgnoresTime_Test: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.sensor.internal.voxel_filter_test.dir/cartographer/sensor/internal/voxel_filter_test.cc.o:voxel_filter_test.cc:typeinfo for cartographer::sensor::(anonymous namespace)::VoxelFilterTest_ReturnsTheFirstPointInEachVoxel_Test: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.sensor.internal.voxel_filter_test.dir/cartographer/sensor/internal/voxel_filter_test.cc.o:voxel_filter_test.cc:vtable for cartographer::sensor::(anonymous namespace)::VoxelFilterTest_HandlesLargeCoordinates_Test: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.sensor.internal.voxel_filter_test.dir/cartographer/sensor/internal/voxel_filter_test.cc.o:voxel_filter_test.cc:vtable for cartographer::sensor::(anonymous namespace)::VoxelFilterTest_HandlesLargeCoordinates_Test: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.sensor.internal.voxel_filter_test.dir/cartographer/sensor/internal/voxel_filter_test.cc.o:voxel_filter_test.cc:vtable for cartographer::sensor::(anonymous namespace)::VoxelFilterTest_IgnoresTime_Test: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.sensor.internal.voxel_filter_test.dir/cartographer/sensor/internal/voxel_filter_test.cc.o:voxel_filter_test.cc:vtable for cartographer::sensor::(anonymous namespace)::VoxelFilterTest_IgnoresTime_Test: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.sensor.internal.voxel_filter_test.dir/cartographer/sensor/internal/voxel_filter_test.cc.o:voxel_filter_test.cc:vtable for cartographer::sensor::(anonymous namespace)::VoxelFilterTest_ReturnsTheFirstPointInEachVoxel_Test: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.sensor.internal.voxel_filter_test.dir/cartographer/sensor/internal/voxel_filter_test.cc.o:voxel_filter_test.cc:vtable for cartographer::sensor::(anonymous namespace)::VoxelFilterTest_ReturnsTheFirstPointInEachVoxel_Test: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.sensor.map_by_time_test.dir/cartographer/sensor/map_by_time_test.cc.o:map_by_time_test.cc:typeinfo for cartographer::sensor::(anonymous namespace)::MapByTimeTest_AppendAndViewTrajectory_Test: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.sensor.map_by_time_test.dir/cartographer/sensor/map_by_time_test.cc.o:map_by_time_test.cc:typeinfo for cartographer::sensor::(anonymous namespace)::MapByTimeTest_TrimmingDoesNotCreateTrajectory_Test: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.sensor.map_by_time_test.dir/cartographer/sensor/map_by_time_test.cc.o:map_by_time_test.cc:typeinfo for cartographer::sensor::(anonymous namespace)::MapByTimeTest_Trimming_Test: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.sensor.map_by_time_test.dir/cartographer/sensor/map_by_time_test.cc.o:map_by_time_test.cc:vtable for cartographer::sensor::(anonymous namespace)::MapByTimeTest_AppendAndViewTrajectory_Test: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.sensor.map_by_time_test.dir/cartographer/sensor/map_by_time_test.cc.o:map_by_time_test.cc:vtable for cartographer::sensor::(anonymous namespace)::MapByTimeTest_AppendAndViewTrajectory_Test: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.sensor.map_by_time_test.dir/cartographer/sensor/map_by_time_test.cc.o:map_by_time_test.cc:vtable for cartographer::sensor::(anonymous namespace)::MapByTimeTest_TrimmingDoesNotCreateTrajectory_Test: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.sensor.map_by_time_test.dir/cartographer/sensor/map_by_time_test.cc.o:map_by_time_test.cc:vtable for cartographer::sensor::(anonymous namespace)::MapByTimeTest_TrimmingDoesNotCreateTrajectory_Test: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.sensor.map_by_time_test.dir/cartographer/sensor/map_by_time_test.cc.o:map_by_time_test.cc:vtable for cartographer::sensor::(anonymous namespace)::MapByTimeTest_Trimming_Test: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.sensor.map_by_time_test.dir/cartographer/sensor/map_by_time_test.cc.o:map_by_time_test.cc:vtable for cartographer::sensor::(anonymous namespace)::MapByTimeTest_Trimming_Test: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.sensor.point_cloud_test.dir/cartographer/sensor/point_cloud_test.cc.o:point_cloud_test.cc:typeinfo for cartographer::sensor::(anonymous namespace)::PointCloudTest_TransformPointCloud_Test: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.sensor.point_cloud_test.dir/cartographer/sensor/point_cloud_test.cc.o:point_cloud_test.cc:typeinfo for cartographer::sensor::(anonymous namespace)::PointCloudTest_TransformTimedPointCloud_Test: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.sensor.point_cloud_test.dir/cartographer/sensor/point_cloud_test.cc.o:point_cloud_test.cc:vtable for cartographer::sensor::(anonymous namespace)::PointCloudTest_TransformPointCloud_Test: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.sensor.point_cloud_test.dir/cartographer/sensor/point_cloud_test.cc.o:point_cloud_test.cc:vtable for cartographer::sensor::(anonymous namespace)::PointCloudTest_TransformPointCloud_Test: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.sensor.point_cloud_test.dir/cartographer/sensor/point_cloud_test.cc.o:point_cloud_test.cc:vtable for cartographer::sensor::(anonymous namespace)::PointCloudTest_TransformTimedPointCloud_Test: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.sensor.point_cloud_test.dir/cartographer/sensor/point_cloud_test.cc.o:point_cloud_test.cc:vtable for cartographer::sensor::(anonymous namespace)::PointCloudTest_TransformTimedPointCloud_Test: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.transform.rigid_transform_test.dir/cartographer/transform/rigid_transform_test.cc.o:rigid_transform_test.cc:typeinfo for cartographer::transform::(anonymous namespace)::RigidTransformTest<double>: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.transform.rigid_transform_test.dir/cartographer/transform/rigid_transform_test.cc.o:rigid_transform_test.cc:typeinfo for cartographer::transform::(anonymous namespace)::RigidTransformTest<float>: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.transform.rigid_transform_test.dir/cartographer/transform/rigid_transform_test.cc.o:rigid_transform_test.cc:vtable for cartographer::transform::(anonymous namespace)::RigidTransformTest<float>: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.transform.rigid_transform_test.dir/cartographer/transform/rigid_transform_test.cc.o:rigid_transform_test.cc:vtable for cartographer::transform::(anonymous namespace)::RigidTransformTest<float>: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.transform.rigid_transform_test.dir/cartographer/transform/rigid_transform_test.cc.o:rigid_transform_test.cc:vtable for cartographer::transform::(anonymous namespace)::RigidTransformTest_Identity2DTest_Test<float>: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.transform.rigid_transform_test.dir/cartographer/transform/rigid_transform_test.cc.o:rigid_transform_test.cc:vtable for cartographer::transform::(anonymous namespace)::RigidTransformTest_Identity2DTest_Test<float>: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.transform.rigid_transform_test.dir/cartographer/transform/rigid_transform_test.cc.o:rigid_transform_test.cc:vtable for cartographer::transform::(anonymous namespace)::RigidTransformTest_Identity3DTest_Test<float>: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.transform.rigid_transform_test.dir/cartographer/transform/rigid_transform_test.cc.o:rigid_transform_test.cc:vtable for cartographer::transform::(anonymous namespace)::RigidTransformTest_Identity3DTest_Test<float>: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.transform.rigid_transform_test.dir/cartographer/transform/rigid_transform_test.cc.o:rigid_transform_test.cc:vtable for cartographer::transform::(anonymous namespace)::RigidTransformTest_Inverse2DTest_Test<float>: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.transform.rigid_transform_test.dir/cartographer/transform/rigid_transform_test.cc.o:rigid_transform_test.cc:vtable for cartographer::transform::(anonymous namespace)::RigidTransformTest_Inverse2DTest_Test<float>: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.transform.transform_interpolation_buffer_test.dir/cartographer/transform/transform_interpolation_buffer_test.cc.o:transform_interpolation_buffer_test.cc:typeinfo for cartographer::transform::(anonymous namespace)::TransformInterpolationBufferTest_testHas_Test: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.transform.transform_interpolation_buffer_test.dir/cartographer/transform/transform_interpolation_buffer_test.cc.o:transform_interpolation_buffer_test.cc:typeinfo for cartographer::transform::(anonymous namespace)::TransformInterpolationBufferTest_testLookupSingleTransform_Test: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.transform.transform_interpolation_buffer_test.dir/cartographer/transform/transform_interpolation_buffer_test.cc.o:transform_interpolation_buffer_test.cc:typeinfo for cartographer::transform::(anonymous namespace)::TransformInterpolationBufferTest_testLookup_Test: error: undefined reference to 'typeinfo for testing::Test'
CMakeFiles/cartographer.transform.transform_interpolation_buffer_test.dir/cartographer/transform/transform_interpolation_buffer_test.cc.o:transform_interpolation_buffer_test.cc:vtable for cartographer::transform::(anonymous namespace)::TransformInterpolationBufferTest_testHas_Test: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.transform.transform_interpolation_buffer_test.dir/cartographer/transform/transform_interpolation_buffer_test.cc.o:transform_interpolation_buffer_test.cc:vtable for cartographer::transform::(anonymous namespace)::TransformInterpolationBufferTest_testHas_Test: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.transform.transform_interpolation_buffer_test.dir/cartographer/transform/transform_interpolation_buffer_test.cc.o:transform_interpolation_buffer_test.cc:vtable for cartographer::transform::(anonymous namespace)::TransformInterpolationBufferTest_testLookupSingleTransform_Test: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.transform.transform_interpolation_buffer_test.dir/cartographer/transform/transform_interpolation_buffer_test.cc.o:transform_interpolation_buffer_test.cc:vtable for cartographer::transform::(anonymous namespace)::TransformInterpolationBufferTest_testLookupSingleTransform_Test: error: undefined reference to 'testing::Test::TearDown()'
CMakeFiles/cartographer.transform.transform_interpolation_buffer_test.dir/cartographer/transform/transform_interpolation_buffer_test.cc.o:transform_interpolation_buffer_test.cc:vtable for cartographer::transform::(anonymous namespace)::TransformInterpolationBufferTest_testLookup_Test: error: undefined reference to 'testing::Test::SetUp()'
CMakeFiles/cartographer.transform.transform_interpolation_buffer_test.dir/cartographer/transform/transform_interpolation_buffer_test.cc.o:transform_interpolation_buffer_test.cc:vtable for cartographer::transform::(anonymous namespace)::TransformInterpolationBufferTest_testLookup_Test: error: undefined reference to 'testing::Test::TearDown()'
collect2: error: ld returned 1 exit status

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 cmake/modules/FindGMock.cmake | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

Index: git/cmake/modules/FindGMock.cmake
===================================================================
--- git.orig/cmake/modules/FindGMock.cmake
+++ git/cmake/modules/FindGMock.cmake
@@ -20,6 +20,14 @@ if(NOT GMock_FOUND)
     PATHS
       /usr
   )
+  find_library(GTEST_LIBRARIES
+    NAMES gtest
+    PATH_SUFFIXES lib
+  )
+  find_library(GMOCK_A_LIBRARIES
+    NAMES gmock
+    PATH_SUFFIXES lib
+  )
 
   # Find system-wide installed gmock.
   find_library(GMOCK_LIBRARIES
@@ -83,7 +91,7 @@ if(NOT GMock_FOUND)
 
   # System-wide installed gmock library might require pthreads.
   find_package(Threads REQUIRED)
-  list(APPEND GMOCK_LIBRARIES ${CMAKE_THREAD_LIBS_INIT})
+  list(APPEND GMOCK_LIBRARIES ${GMOCK_A_LIBRARIES} ${GTEST_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
 endif()
 
 include(FindPackageHandleStandardArgs)
